administrator/components/com_templates/src/Model/TemplateModel.php

```diff
@@ -1985,4 +1989,161 @@ protected function checkFormat($ext)
 
 		return in_array(strtolower($ext), $this->allowedFormats);
 	}
+
+	/**
+	 * Method to get a list of all the files to edit in a template's media folder.
+	 *
+	 * @return  array  A nested array of relevant files.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function getMediaFiles()
+	{
+		$result = array();
+		$template = $this->getTemplate();
+
+		if (!isset($template->xmldata))
+		{
+			$template->xmldata = TemplatesHelper::parseXMLTemplateFile($template->client_id === 0 ? JPATH_ROOT : JPATH_ROOT . '/administrator', $template->name);
+		}
+
+		if (!isset($template->xmldata->inheritable) || (isset($template->xmldata->parent) && $template->xmldata->parent === ''))
+		{
+			return [];
+		}
+
+		$app  = Factory::getApplication();
+		$path = Path::clean(JPATH_ROOT . '/media/templates/' . ($template->client_id === 0 ? 'site' : 'administrator') . '/' . $template->element . '/');
+		$this->mediaElement = $path;
+
+		if (!is_writable($path))
+		{
+			$app->enqueueMessage(Text::_('COM_TEMPLATES_DIRECTORY_NOT_WRITABLE'), 'error');
+		}
+
+		if (is_dir($path)) {
+			$result = $this->getDirectoryTree($path);
+		} else {
+			return [];
+		}
+
+		return $result;
```

Comment:
reuse `$result` which is a blank array already.

```suggestion
		if (is_dir($path)) {
			$result = $this->getDirectoryTree($path);
		}

		return $result;
```